### ISSUE TITLE
Fix S3928 FP: Parameters of local functions

### ISSUE THREAD

**Issue by hankovich** [SonarSource/sonar-dotnet#4423]
### Description

S3928 is reported inside local functions during arguments' validation.

### Repro steps

```csharp
private void Foo(int i)
{
    Bar(i);

    void Bar(int j)
    {
        throw new ArgumentOutOfRangeException(nameof(j), j, null); // <- S3928: the parameter name 'j' in not declared in the argument list
    }
}
```

### Expected behavior

The diagnostic is not reported.

### Additional information

Nothing is reported when I use `nameof(i)` (i.e. argument name from a containing method).

### Related information

* SonarAnalyzer.CSharp 8.2.0-RC1

**Comment by pavel-mikula-sonarsource** [comment]
Hi @hankovich ,

Thank you for reporting this case, I can confirm it as False Positive.
